lib/smblib/exper.c

```diff
@@ -338,8 +338,8 @@ SMB_Tree_Handle SMB_Logon_And_TCon(SMB_Handle_Type Con_Handle,
 
     if (Con_Handle -> first_tree == NULL) {
 
-        Con_Handle -> first_tree == tree;
-        Con_Handle -> last_tree == tree;
+        Con_Handle -> first_tree = tree;
+        Con_Handle -> last_tree = tree;
 
```

Comment:
@kinkie, after recent commit 8f2566c7, I can no longer find any smblib users in Squid sources? Can we remove smblib now?

N.B. I would still merge this PR (if/when it is ready), so that the fix can be backported easily, and then merge a separate smblib removal PR (if it is possible to remove that library and the code it drags into Squid).

Reply:
backport should not be an issue, this code is not touched since v3 days.

Since v6 needs a manual PR for backport anyway IMHO this PR should be replaced with one proposing a merge to v6 and the v7/v8 just drop smblib entirely.

Reply:
I fully support dropping smblib entirely.

Reply:
It seems to actually ne possible.
PR #2138